File: 14-Facade/src/go/theater_lights.go
package main

import "fmt"

// Subsystem type — the dimmable theater lights.
type TheaterLights struct{}

func (t TheaterLights) On()          { fmt.Println("Theater lights on") }
func (t TheaterLights) Dim(level int) { fmt.Printf("Theater lights dimming to %d%%\n", level) }

File: 14-Facade/src/java/chapter14/TheaterLights.java
package chapter14;

/**
 * Subsystem class — the dimmable theater lights.
 */
public class TheaterLights {
    public void on() {
        System.out.println("Theater lights on");
    }

    public void dim(int level) {
        System.out.println("Theater lights dimming to " + level + "%");
    }
}
